### ISSUE TITLE
Build nodejs 14.16.0 binary from source

### ISSUE THREAD

**Issue by theaquarian007** [vercel/pkg-fetch#116]
I want to build nodejs 14.16.0 binary from source. However, I want to update the node source code before I build it. How can I do that using pkg-fetch. Every time I try, it builds node 14.14.0 and fetches the code from github. I want to point it to the local source code. There is very little documentation on using pkg-fetch.

**Comment by saikoux12** [comment]
want too...

**Comment by b00tsy** [comment]
@theaquarian007: How do you build from source and where do you find the `14.14.0` patch file? I can only see the `14.4.0` path file in /patches.

**Comment by robertsLando** [comment]
In order to have the 14.14.0 you firstly need to create the patch file...

**Comment by robertsLando** [comment]
>  and how one can build one's own base binary until pkg-fetch becomes an official new release...

You can find this explained in my `pkg-binaries` repo. BTW what I do is to place the binaries in the pkg-cache path so when pkg runs it doesn't have to build nodejs from source

**Comment by b00tsy** [comment]
yeah I remember using the steps of paragraph `Usage` described in your README.md to achieve that with the binaries from  https://github.com/jesec/pkg-fetch (I need `linux-x64`), but I couldn't find out what the difference between `uploaded` and `fetched` was. Obviously not just renaming uploaded with fetched ...

**Comment by robertsLando** [comment]
> Obviously not just renaming uploaded with fetched ...

Lol that's the real different, just the name :) Them are the same file but pkg looks for `fetched*` in PKG_CACHE, it is named `uploaded-*` in the resources

**Comment by b00tsy** [comment]
really? 

I gave that a try recently before you commented: downloaded eg. https://github.com/jesec/pkg-fetch/releases/download/v2.7/uploaded-v2.7-node-v14.16.0-linux-x64  renamed it to `fetched-v14.16.0-linux-x64` moved it to `~/.pkg-cache/v2.6` (where other bins eg. fetched-v14.4.0-linux-x64 still exist) and built a project with pkg, but process.version of the built app was still 14.4.0... 

If you say that theoretically should work I'll give it another try...

**Comment by robertsLando** [comment]
> If you say that theoretically should work I'll give it another try...

It works, I'm 100% sure about this as it is what I'm doing all times to build arm binaries

**Comment by theaquarian007** [comment]
I uploaded the source code to aws and pointed the nodeRepo variable to the aws repo. That way I got pkg-fetch to get the codebase from aws. However, the patching is still failing. I tried using the patch file from https://github.com/jesec/pkg-fetch/tree/master/patches. However, I get the following error:

> Applying patches
patch: **** Only garbage was found in the patch input. 

Is there a valid patch file available?
